identity/backend/src/main/java/io/camunda/identity/usermanagement/repository/UserRepository.java

```diff
@@ -24,6 +31,52 @@ public UserRepository(final JdbcTemplate jdbcTemplate) {
 
   public List<CamundaUser> loadUsers() {
     return jdbcTemplate.query(
-        DEF_USERS_QUERY, (rs, rowNum) -> new CamundaUser(rs.getString(1), rs.getBoolean(2)));
+        DEF_USERS_QUERY,
+        (rs, rowNum) ->
+            new CamundaUser(rs.getInt(1), rs.getString(2), rs.getString(4), rs.getBoolean(3)));
+  }
+
+  public void createProfile(final CamundaUser user) {
+    final var id =
+        jdbcTemplate.queryForObject(
+            DEF_USER_ID_QUERY, new Object[] {user.username()}, Integer.class);
+    jdbcTemplate.update(
+        DEF_PROFILE_CREATE,
+        (ps) -> {
+          ps.setInt(1, id);
+          ps.setString(2, user.email());
+        });
+  }
+
+  public CamundaUser loadUser(final String username) {
+    return jdbcTemplate
+        .query(
+            DEF_USER_QUERY,
+            new Object[] {username},
+            (rs, rowNum) ->
+                new CamundaUser(rs.getInt(1), rs.getString(2), rs.getString(4), rs.getBoolean(3)))
+        .get(0);
+  }
+
+  public void updateProfile(final CamundaUser user) {
+    final var id =
+        jdbcTemplate.queryForObject(
```

Comment:
❓ Instead of making two calls to the database can we use a single statement with a join to limit result by username?